### PR TITLE
Date type support

### PR DESCRIPTION
**Pull request by msangel**
Changes:
* java support moved from 1.7 to 1.8. Sorry.
* because of incompatible changes, the library version now is 0.8.x
* documentation upgrade to reflect most important changes
* time-related changes:
  * for propper support java8 datetime types by the jackson the `jackson-datatype-jsr310 dependency` added 
  * `strftime` formatter rewritten from scratch and moved to own library(`ua.co.k:strftime4j:1.0.3`), so can be upgradable independently via maven
  * datetime types now printed in the output with the same format as it would be in ruby, so it's +1 for compatibility
  * the template now can have default `locale` so the formatting of some variables will be locale-specific(most obvious place - timezone names and month names in different languages)
  * the template now can have default `timezone`, so in all place when the timezone was printed but not provided - the default one will be used  
  * java [built-in plugin system ](https://www.oracle.com/technical-resources/articles/javase/extensible.html) is used for supporting different time libraries (like [joda-time](https://www.joda.org/joda-time/) or [ThreeTen](https://www.threeten.org/)). See example for java7 types in `Java7DateTypesSupport` class.
  * date filter now handle dates with 'T' part, fix this: https://github.com/bkiers/Liqp/issues/171
 * other minor changes:
   * `Abs`, `Minus`, `Plus`, `Modulo` and `Times` filters fixed behavior with handling numbers with decimal points(now uses the same number format as ruby's `float` type)
   * `Abs` filter fixed behavior when input is not a number (was - return input, propper - return `0`)
   * `Size` filer now can count the size of objects (key-value pairs)
   * fix `empty` and `blank` objects/keywords rendering - now as empty line(before - it was as "Object@2a139a55")
   * the expressions with `>`, `<`, `>=` and `<=` now handle `Comparable` objects if they can be compared.
   * in case of error of accessing object by square brackets the error now show real error expression, like `a[b]` instead of `a[null]`

### CODE REVIEW

**Comment by msangel** [comment]
@bkiers the work is done on the datetime support, I believe we can release `0.8.0-a1`. Still, as the changeset is huge, please review it and I will answer all questions.

**Comment by bkiers** [comment]
Nice @msangel, I'll try to go through it this week 👍

File: src/main/java/liqp/filters/Abs.java [hunk 1]
```diff
@@ -10,9 +29,9 @@ public Object apply(Object value, Object... params) {
         }
 
         if (super.isNumber(value) || super.canBeDouble(value)) {
-            return Math.abs(super.asNumber(value).doubleValue());
+            return asFormattedNumber(new BigDecimal(super.asNumber(value).toString()).abs());
         }
 
-        return value;
+        return 0;
```
Comment (bkiers) [inline src/main/java/liqp/filters/Abs.java:35]: Good to add the reference source, else I'd thought the `return 0;`  was incorrect :+1:

**Review by bkiers — APPROVED** [review]
Apologies for the late follow up... The PR looks great: well tested and documented :tada: 

Feel free to merge.

**Comment by bkiers** [comment]
Oh, with all the time and effort you've put in this (and other PR's!), do please also add your name to the `<developers>` in the POM file @msangel

**Pull request merged**